spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/mapper/JavaBeanColumnMapper.scala

```diff
@@ -28,6 +34,24 @@ class JavaBeanColumnMapper[T : ClassTag](columnNameOverride: Map[String, String]
     method.getParameterTypes.size == 1 &&
     method.getReturnType == Void.TYPE
 
+  private def annotationForFieldName(fieldName: String): String = {
```

Comment:
This may just be the monad lover in me, but if this returned Option[String]

We could be a little more functional and the above code for field name could be

`annotationName getOrElse fieldName`

Also this bit can be changed to

```
val maybeField = Try(cls.getField(fieldName))
maybeField.toOption.flatMap {
  case f =>
    if (f.isAnnotationPresent(classOf[Column])) {
      Some(f.getAnnotation(classOf[Column]).name())
  } else if (f.isAnnotationPresent(classOf[Field])) {
    Some(f.getAnnotation(classOf[Field]).name()) 
  } else None
}
```

You could also probably change the internal calls to 
`(Try(f.getAnnotation(classOf[Column]).name()) orElse Try(f.getAnnotation(classOf[Field]).name())).toOption`

Which would change the whole thing to

```
val maybeField = Try(cls.getField(fieldName))
maybeField.flatMap( f => 
  (
    Try(f.getAnnotation(classOf[Column]).name())  orElse 
    Try(f.getAnnotation(classOf[Field]).name())
   ).toOption
```